containers/go/.devcontainer/Dockerfile

```diff
@@ -49,16 +49,19 @@ RUN apt-get update \
         github.com/mgechev/revive@latest  \
         github.com/go-delve/delve/cmd/dlv@latest 2>&1 \
     #
-    # Install Go tools w/o module support
-    && go get -v github.com/alecthomas/gometalinter 2>&1 \
+    # Build Go tools w/o module support
+    && GOPATH=/tmp/gotools go get -v github.com/alecthomas/gometalinter 2>&1 \
     #
-    # Install gocode-gomod
-    && go get -x -d github.com/stamblerre/gocode 2>&1 \
-    && go build -o gocode-gomod github.com/stamblerre/gocode \
-    && mv gocode-gomod $GOPATH/bin/ \
+    # Build gocode-gomod
+    && GOPATH=/tmp/gotools go get -x -d github.com/stamblerre/gocode 2>&1 \
+    && GOPATH=/tmp/gotools go build -o gocode-gomod github.com/stamblerre/gocode \
+    #
+    # Install Go tools
+    && mv /tmp/gotools/bin/* /usr/local/bin/ \
```

Comment:
This seems to be generally a nice way of handling this! However, why not move these to /go/bin ?  One thing I do notice is that the Go extension will update the GOPATH location under certain circumstances, so if it's somewhere else, you'd end up having two different versions on disk.

Reply:
> However, why not move these to /go/bin ?

Leaving anything in `/go/bin` defeats the whole purpose of this PR: to make it possible to work with the image as a non-root user. Consider a situation, when I want to install a newer version of the same tool. If it were in `/go/bin` it would be owned by `root` and so I won't be able to overwrite it. Exactly the thing that happened with the package cache.

Also, `/usr/local/bin` is the [standard FHS location](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch04s09.html) for installing the binaries which are not maintained by the package manager, while `/go/bin` is some random path that upstream authors decided to use for the `golang:1` image.

> One thing I do notice is that the Go extension will update binaries in GOPATH location under certain circumstances (e.g. if you switch to gopls the first time), so if it's somewhere else, you'd end up having two different versions on disk.

I think having two versions of the same tool is perfectly fine. That is what you usually do when the package manager provides an ancient version of the tool that you need. You just install the new version to your `$HOME/.local/bin` and enjoy the cool features 😄 .

In fact, `$PATH` within the image is set to `/go/bin:/usr/local/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin`. So if we put all the tools to `/usr/local/bin` and VS Code decides to install something to `$GOPATH` it will take precedence and everything will work smoothly.

Reply:
Apologies for the delayed response.  I was out of office last week.

I think we'll also need to chown or chmod the `/go` folder anyway won't we?  Anything the user installs ad-hoc will still land in that folder.  We could just `chown -R vscode /go`.

Reply:
The changed `Dockerfile` doesn't write anything to the `/go` directory.

Now let's take a look at what upstream `golang:1` image does to permissions for `/go`:

```
$ docker run --rm golang:1 ls -ld /go
drwxrwxrwx 4 root root 4096 Feb 14 01:21 /go
$ docker run --rm golang:1 ls -lR /go
/go:
total 8
drwxrwxrwx 2 root root 4096 Feb 14 01:21 bin
drwxrwxrwx 2 root root 4096 Feb 14 01:21 src

/go/bin:
total 0

/go/src:
total 0
```

As you can see the upstream image allows everyone to write to `/go` and its subdirectories, so no `chrown` or `chmod` is necessary.

Reply:
Good catch.  Ok, LGTM!